src/test/java/plugins/WorkflowPluginTest.java

```diff
@@ -158,4 +168,31 @@
         build.shouldContainsConsoleOutput("divided into 3 sets");
     }
 
+    @WithDocker
+    @WithPlugins({"workflow-aggregator@1.14", "docker-workflow@1.14-SNAPSHOT", "git", "ssh-agent@1.10-SNAPSHOT"})
+    @WithCredentials(credentialType=WithCredentials.SSH_USERNAME_PRIVATE_KEY, values={"git", "/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/unsafe"}, id="gitcreds")
+    @Issue("JENKINS-27152")
+    @Test public void sshGitInsideDocker() throws Exception {
+        // Pending https://github.com/jenkinsci/docker-workflow-plugin/pull/31 (and additional APIs in DockerFixture etc.)
+        // it is not possible to run this on a bind-mounted Docker slave, so we cannot verify that paths are local to the slave.
+        GitContainer container = gitServer.get();
+        String host = container.host();
+        int port = container.port();
+        GitRepo repo = new GitRepo();
+        repo.commit("Initial commit");
+        repo.transferToDockerContainer(host, port);
+        WorkflowJob job = jenkins.jobs.create(WorkflowJob.class);
+        job.script.set(
+            "node {ws('" + tmp.getRoot() + "') {\n" + // TODO UNIX_PATH_MAX workaround
```

Comment:
Not happy about this but not sure what else to do @cyrille-leclerc: if you have a long workspace path, putting the SSH agent socket in or near it will not work, so the SSH Agent plugin will fall back to a global temp dir that will not be accessible from inside the Docker container.

Once https://github.com/jenkinsci/jenkins/pull/2066 makes it into an LTS and we call it from both sides, that method could be adjusted to give an option to use a shortened path, say `/tmp/<SHA-1-of-workspace>`, which would be deterministic and unique, but nowhere near the workspace. Or `/<slave-FS-root>/<some-hash>`, to retain the property of being under the slave FS root and thus amenable to cleanup, bulk `chmod` operations, etc.?